editor/plugins/script_text_editor.cpp

```diff
@@ -1726,7 +1726,15 @@ void ScriptTextEditor::drop_data_fw(const Point2 &p_point, const Variant &p_data
 
 				String variable_name = String(node->get_name()).to_snake_case().validate_identifier();
 				if (use_type) {
-					text_to_drop += vformat("@onready var %s: %s = %s%s\n", variable_name, node->get_class_name(), is_unique ? "%" : "$", path);
+					String class_name = node->get_class_name();
+					Ref<Script> node_script = node->get_script_instance()->get_script();
+					if (node_script.is_valid()) {
+						StringName global_node_script_name = node_script->get_global_name();
+						if (global_node_script_name != "") {
```

Comment:
```suggestion
						if (!global_node_script_name.is_empty()) {
```
Preferred style for checking empty strings.

Reply:
Ah, it's a StringName, sorry. Then it should be `if (global_node_script_name != StringName()).

Reply:
Without ! at the beginning.